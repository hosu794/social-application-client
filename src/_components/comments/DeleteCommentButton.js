import React from "react";

import { commentActions } from "../../_actions";

import { useDispatch } from "react-redux";

function DeleteCommentButton({ handleDeleteComment }) {
  return (
    <button onClick={handleDeleteComment} className="button is-danger">
      Delete
    </button>
  );
}

export default DeleteCommentButton;
