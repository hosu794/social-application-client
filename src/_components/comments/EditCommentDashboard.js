import React, { useEffect } from "react";
import { useSelector, useDispatch } from "react-redux";
import ReactDOM from "react-dom";
import RichTextEditor from "../stories/TextEditor/RichTextEditor";
import { Formik, Field, Form as FormikForm, ErrorMessage } from "formik";
import { ConfigProvider, Radio, Button, Typography } from "antd";
import "react-quill/dist/quill.core.css";
import "react-quill/dist/quill.snow.css";
import "antd/dist/antd.css";
import * as Yup from "yup";
import "react-quill/dist/quill.core.css";
import "react-quill/dist/quill.snow.css";
import "antd/dist/antd.css";
import { commentActions, alertActions } from "../../_actions";
import PropTypes from "prop-types";
import { IComment } from "../../_types";

function EditCommentDashboard(props) {
  const dispatch = useDispatch();

  const currentComment = useSelector((state) => state.comment.currentComment);
  const loaded = useSelector((state) => state.comment.loaded);
  const alertType = useSelector((state) => state.alert.type);
  const isUpdatedSuccess = alertType === "alert-success";

  const loadCurrentComment = async () => {
    await dispatch(commentActions.getCommentById(props.match.params.id));
  };

  useEffect(() => {
    dispatch(alertActions.clear());
    loadCurrentComment();
  }, []);

  return (
    <section>
      <h1 className="title">Edit comment</h1>
      {loaded ? (
        <Formik
          initialValues={{
            body: currentComment.body,
          }}
          validationSchema={Yup.object().shape({
            body: Yup.string().required("body required"),
          })}
          onSubmit={(values, { setSubmitting }) => {
            setTimeout(() => {
              dispatch(
                commentActions.updateComment(
                  {
                    body: values.body,
                    storyId: currentComment.storyResponse.id,
                  },
                  currentComment.id
                )
              );
              setSubmitting(false);
            }, 300);
          }}
        >
          {({
            isSubmitting,
            isValid,
            setFieldValue,
            values,
            handleChange,
            handleBlur,
            touched,
            errors,
          }) => {
            return (
              <FormikForm>
                <div className="form-group">
                  <label htmlFor="body">Body</label>
                  <Field
                    name="body"
                    type="text"
                    className={
                      "input" +
                      (errors.body && touched.body ? " is-danger" : "")
                    }
                  />
                  <ErrorMessage
                    name="body"
                    component="div"
                    className="help is-danger"
                  />
                </div>
                <div
                  style={{
                    paddingBottom: 24,
                    marginTop: 16,
                  }}
                >
                  <div className="level">
                    <div className="level-item level-center">
                      <button
                        className="button is-info is-medium"
                        htmlType="submit"
                        type="submit"
                      >
                        Edit
                      </button>
                    </div>
                  </div>
                </div>
              </FormikForm>
            );
          }}
        </Formik>
      ) : (
        "Comment not found"
      )}
      {isUpdatedSuccess ? (
        <div className="notification is-info">Comment Updated Successfully</div>
      ) : null}
    </section>
  );
}

EditCommentDashboard.propTypes = {
  currentComment: PropTypes.instanceOf(IComment),
  loaded: PropTypes.bool,
  alertType: PropTypes.string,
};

EditCommentDashboard.defaultProps = {
  currentComment: undefined,
  loaded: false,
  alertType: "",
};

export default EditCommentDashboard;
