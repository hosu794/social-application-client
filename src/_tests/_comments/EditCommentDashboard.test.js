import React from "react";

import { Provider } from "react-redux";

import { mount } from "enzyme";

import {
  mockStory,
  mockComment,
  store,
  mockIdMatchParams,
} from "../_testHelpers";
import DeleteCommentButton from "../../_components/comments/DeleteCommentButton";
import { MemoryRouter } from "react-router-dom";
import EditCommentDashboard from "../../_components/comments/EditCommentDashboard";

describe("Tests for the EditCommentDashboard Component", () => {
  let TestingComponent = () => (
    <Provider store={store}>
      <MemoryRouter>
        <EditCommentDashboard match={mockIdMatchParams} />
      </MemoryRouter>
    </Provider>
  );

  store.dispatch = jest.fn();

  let wrapper = mount(<TestingComponent />);

  test("should render componenent correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should component be defined", () => {
    expect(EditCommentDashboard).toBeDefined();
  });

  test("should display Notification about successfully updated comment", () => {
    console.log(wrapper.debug());

    let element;

    const expectedText = "Comment Updated Successfully";

    element = wrapper.find(".notification").text();

    expect(element).toEqual(expectedText);
  });

  test("should update an password input when it is changed", async () => {
    let expectedValue = "Some fancy comment";

    await wrapper
      .find("input")
      .first()
      .simulate("change", {
        persist: () => {},
        target: {
          name: "body",
          value: expectedValue,
        },
      });

    let newValue = wrapper
      .find(EditCommentDashboard)
      .find("input")
      .first()
      .props().value;

    expect(newValue).toEqual(expectedValue);

    await wrapper.find(".button").simulate("click");

    expect(store.dispatch.mock.calls.length).toEqual(2);
  });
});
