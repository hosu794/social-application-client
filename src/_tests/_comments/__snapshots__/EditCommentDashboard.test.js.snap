// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for the EditCommentDashboard Component should render componenent correctly 1`] = `ReactWrapper {}`;
